src/common/outputService/outputService.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export type OutputFlags = {
+  verbosity?: boolean;
+  consice?: boolean;
+};
+
+/**
+ * Service interface to print the output
+ *
+ * @author JuanStenghele-sf
+ */
+export interface OutputService {
+  /**
+   * Prints a summary of the operation being done
+   */
+  printOpSummary(): Promise<void>;
```

Comment:
Two questions regarding this method:
- Could it have a more generic name, like print()? In the future we may have commands that need to print stuff that is not necessarily a summary. 
- If its main responsibility is to print a message, why it returns a promise? Does it need to asynchronous?

Reply:
- The idea of the method is to print a summary of the operation done, so that's why it has summary in its name. The idea would be in the future with more output services add more methods (if it is not a summary).
- It is async because we gather the information needed for the summary in that same method. (We end up doing some queries here)

Reply:
Thanks for the answer.

- Got it.
- IMO the return type in the interface shouldn't be a promise, that is an implementation detail that the concrete classes should define. Otherwise all implementations are required to return a promise even if it's not needed, for example ResumeCommandOutputService.printOpSummary(). I think is safe to make the interface and abstract classes to return void and the concrete classes like DeploySummaryOutputService and AbstractPromoteOutputService to return Promise<void>. Open for discussion.